raiden/messages/metadata.py

```diff
@@ -33,13 +33,13 @@ def _validate_address_metadata(self) -> None:
 
     @cached_property
     def hash(self) -> bytes:
-        return keccak(self._serialize_canonicaljson())
+        return keccak(self._serialize_canonical())
 
-    def _serialize_canonicaljson(self) -> bytes:
-        route = [to_checksum_address(address) for address in self.route]
+    def _serialize_canonical(self) -> bytes:
         if not self.address_metadata:
-            return canonicaljson.encode_canonical_json({"route": route})
+            return rlp.encode(self.route)
```

Comment:
I don't feel good about this: in regards to #7021 we planned for treating the whole `Metadata` container as being hashed via canonical json.
What kind of backwards compatibility are we trying to achieve here?

Reply:
Right now, the light client does not include address_metadata and uses the RLP encoding, since the encoding change has not been coordinated between the clients, AFAIK.

I'm also hesitant to promote the way it was before this PR because:
* it introduces a breaking change (compared to LC and v1.2)
* the signature encoding for the whole `Metadata` object is likely to change again when we want to allow passing through arbitrary metadata

So if we change the signatures, let's do it in a way that also supports signing arbitrary metadata.

Reply:
If we ensure this signature change is future proof (i.e. it'll allow deterministic signature hash for arbitrary `metadata` payloads, even containing unknown fields), I'd say we're in favor of the breaking change, specially for allowing metadata passthrough on mediators, we just need to coordinate on that, since this broke without we noticing.

Reply:
e.g. from what I can see below, this kind of assumes the only 2 fields present are `route` and `address_metadata`, and also encode them in a specific way. Maybe we could go with something more generic, like just feeding the whole object to `encode_canonical_json` and hashing the resulting string?

Reply:
> Maybe we could go with something more generic, like just feeding the whole object to encode_canonical_json and hashing the resulting string?

Yes, this is also what I would suggest, since it would get rid of the different cases and allow arbitrary metadata passthrough. I don't have any experience with canonical JSON myself, but I assume it is well defined and well supported enough.

Reply:
Here's my try to encode the whole metadata with canonical JSON: https://github.com/raiden-network/raiden/pull/7075